.github/actions/build-ee-extra/action.yml

```diff
@@ -0,0 +1,126 @@
+name: Build ee-extra
+description: Build Metabase Enterprise Edition with extra features
+
+inputs:
+  iam-role:
+    description: "The IAM role to assume"
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Prepare Java
+      uses: actions/setup-java@v3
+      with:
+        java-version: '17'
+        distribution: 'temurin'
+    - name: Reveal its original version.properties
+      run: jar xf  ./bin/docker/metabase.jar version.properties && cat version.properties
+      shell: bash
+    - name: get major version
+      id: major_version
+      run: echo "major_version=$(cat version.properties | grep 'tag=' | cut -d'.' -f2)" >> $GITHUB_OUTPUT
+      shell: bash
+    - name: get drivers registry
+      run: |
+        curl -OL https://raw.githubusercontent.com/metabase/metabase-registry/main/registry.yaml
+      shell: bash
+    - uses: actions/setup-node@v4
+      with:
+        node-version: lts/Hydrogen # 18.x.x
+    - run: rm package.json && npm install yaml
+      shell: bash
+    - name: parse drivers registry
+      uses: actions/github-script@v7
+      id: drivers_registry
+      with:
+        script: | # js
+          const { parse } = require('yaml');
+          const fs = require('fs');
+          const registry = parse(fs.readFileSync('registry.yaml', 'utf8'));
+
+          const drivers = registry.drivers;
+          const majorVersion = '${{ steps.major_version.outputs.major_version }}';
+
+          const getDriverUrl = (driverName) => {
+            const driver = drivers.find(driver => driver.name === driverName);
+
+            if (!driver) {
+              throw new Error(`Driver ${driverName} not found in the registry`);
+            }
+
+            const driverUrl = driver.versions[majorVersion] ?? driver.versions.default;
+
+            if (!driverUrl) {
+              throw new Error(`Driver ${driverName} version ${majorVersion} not found in the registry`);
+            }
+
+            return driverUrl;
+          };
+
+          const extraDrivers = {
+            firebolt: getDriverUrl('firebolt'),
+            starburst: getDriverUrl('starburst'),
+            exasol: getDriverUrl('exasol'),
+            exasol_jdbc: drivers.find(d => d.name === 'exasol').deps[0],
+            clickhouse: getDriverUrl('clickhouse'),
+            ocient: getDriverUrl('ocient'),
+            materialize: getDriverUrl('materialize'),
+          };
```

Comment:
i'm not a huge fan of the duplication with ee-extra here. we're definitely gonna forget to update something here at some point.